File: datacenter/src/services/dataCenter/cluster.ts
import { ClusterNode } from "./clusterNode";
import { Threshold } from "./threshold";
import { ClusterStatus } from "./clusterStatus.enum";

export interface Cluster {
    name: string;
    clusterNodes: Array<ClusterNode>;
    thresholds: Array<Threshold>;
    status: ClusterStatus;

}

File: datacenter/src/DataCenterApi/DataCenterApi/Models/Cluster.cs
using System.Collections.Generic;
using System.Linq;

namespace DataCenterApi.Models
{
  public class Cluster
  {
    public string Name { get; set; }
    public List<ClusterNode> Nodes { get; set; }
    public List<Threshold> Thresholds { get; set; }
    public ClusterStatus Status { get; set; }

    public Cluster()
    {
      Nodes = new List<ClusterNode>();
      Thresholds = new List<Threshold>();
    }

    public void SetStatus()
    {
      if (!Nodes.Any())
      {
        Status = ClusterStatus.NoThresholdsSet;
        return;
      }

      Status = (ClusterStatus)Nodes.SelectMany(n => n.Metrics).Select(m => m.Status).Max();
    }
  }
}
